feat: steps support for circular progress bar (#47940)
ant-design/ant-design: components/progress/index.en-US.md

<code src="./demo/linecap.tsx">Stroke Linecap</code>
<code src="./demo/gradient-line.tsx">Custom line gradient</code>
<code src="./demo/steps.tsx">Progress bar with steps</code>

> <code src="./demo/circle-steps.tsx">Circular progress bar whit steps</code>

Reviewer: ```suggestion
<code src="./demo/circle-steps.tsx" version="5.16.0">Circular progress bar whit steps</code>
```

| Property | Description | Type | Default | Version |
| --- | --- | --- | --- | --- |

> | steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |

Reviewer: ```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count: number, gap: number } | - | 5.16.0 |
```

> | steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.16.0 |

Reviewer: ```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |
```

| strokeColor | The color of circular progress, render gradient when passing an object | string \| { number%: string } | - | - |
| strokeWidth | To set the width of the circular progress, unit: percentage of the canvas width | number | 6 | - |

### `type="dashboard"`

| Property | Description | Type | Default | Version |
| --- | --- | --- | --- | --- |

> | steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |

Reviewer: ```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count: number, gap: number } | - | 5.16.0 |
```

> | steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.15.4 |

Reviewer: ```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.16.0 |
```

> | steps | The total step count.When passing an object, `count` refers to the number of steps, and `space` refers to the distance between them.When passing number, the default value for `space` is 2. | number \| { count: number, space: number } | - | 5.16.0 |

Reviewer: ```suggestion
| steps | The total step count.When passing an object, `count` refers to the number of steps, and `gap` refers to the distance between them.When passing number, the default value for `gap` is 2. | number \| { count?: number, gap?: number } | - | 5.16.0 |
```
